runtime/CMakeLists.txt

```diff
@@ -50,6 +50,7 @@ FetchContent_Declare(
 )
 
 function(fetch_pybind11)
+    find_package(Python COMPONENTS Interpreter Development) # Required for non-standard Python system intsalls
```

Comment:
Oh boy I just went through a whole ordeal about this with pybind and how it finds Python 😭

Reply:
Wait, so it wasn't just me and my silly system causing this? Or is this the cause of the problem and I readded it?

Reply:
It might make sense to show you some of the issues I was having after I dig them up, but I think having that command here was causing a problem (or at the very least not solving the issue I had).

Reply:
Ah ok, gotcha. Without this line my system outright fails to build the runtime --- will explore alternatives, or just drop to a ML2_"* docker and validate it on there

Reply:
I'm happy to put it there if it works, will just have to validate on all configurations we currently use. Just found this:
https://github.com/PennyLaneAI/catalyst/pull/669/commits/a5e55cd5392806c7596ade250939acdacad951d6

Reply:
See also this section in the pybind docs which mention that `Development` is not available, but that you should be able to use `Development.Module` instead:
https://pybind11.readthedocs.io/en/stable/compiling.html#findpython-mode

Reply:
I wonder why this had been removed recently in this PR https://github.com/PennyLaneAI/catalyst/pull/669/files#diff-f55f8aeab7c98a234b9dcc445f57ae2a1cb43ba243d1dc489e3f5b20ba1b7e11L25 ?!

Reply:
The commit mentions why it was removed

Reply:
I pulled it out of here again. Assuming the runtime can work without it ---- it seems to be fine on my UB2204 machine, but may be a problem later (cannot build without in on Arch derivatives). Assuming that's fine.